src/plugins/intel_gpu/src/kernel_selector/kernels/gather/gather_kernel_ref.cpp

```diff
@@ -253,7 +253,7 @@ JitConstants GatherKernelRef::GetJitConstants(const gather_params& params) const
     if (!dyn_gather_idx_dim)
         jit.AddConstant(MakeJitConstant("AXIS_DIM", GetGatherMaxIndexDim(params)));
 
-    if (params.is_shape_agnostic)
+    if (params.is_shape_agnostic && params.inputs[0].is_dynamic())
```

Comment:
This change was already merged to the master branch. (https://github.com/openvinotoolkit/openvino/pull/23548)